internal/fs/fs.go

```diff
@@ -2077,13 +2077,31 @@ func (fs *fileSystem) WriteFile(
 	defer in.Unlock()
 
 	// Serve the request.
-	if err := in.Write(ctx, op.Data, op.Offset); err != nil {
-		return err
+	if fs.shouldTriggerBufferEnabledWrite(in) {
+		// Trigger write file buffering flow.
+		err := in.WriteWithBuffer(fs.mountConfig.WriteConfig.BufferSizeMB, op.Data, op.Offset)
+		if err != nil {
+			return err
+		}
+	} else {
+		// Trigger the temp file flow.
+		if err := in.Write(ctx, op.Data, op.Offset); err != nil {
+			return err
+		}
 	}
 
 	return
 }
 
+func (fs *fileSystem) shouldTriggerBufferEnabledWrite(in *inode.FileInode) bool {
+	if fs.mountConfig.WriteConfig.EnableStreamingWrites &&
+		!fs.mountConfig.WriteConfig.CreateEmptyFile &&
```

Comment:
do we need these two conditions?
in.IsLocal is true only when createEmptyFile=false.

Reply:
removed 	`!fs.mountConfig.WriteConfig.CreateEmptyFile` condition